packages/amplify-e2e-tests/src/__tests__/tags.test.ts

```diff
@@ -58,7 +58,7 @@ describe('generated tags test', () => {
 function checkEquality(localTags: {}[], generatedTags: {}[]) {
   localTags.forEach((tagObj) => {
     const rootTag = generatedTags.find((obj) => obj['Key'] === tagObj['Key']);
-    if (tagObj['Key'] !== rootTag['Key']) return false;
+    return tagObj['Key'] === rootTag['Key'];
```

Comment:
This function is fundamentally broken. The function always returns true, because forEach is itself a function.

I understand it's outside the scope of the PR, but we need to audit every test that uses this.

Reply:
Fortunatelly it's not a big blast radious.
![image](https://github.com/aws-amplify/amplify-cli/assets/5849952/194c7d25-3b41-411e-bd2b-9f566aab4fae)